pytorch_lightning/trainer/predict_loop.py

```diff
@@ -51,6 +51,10 @@ def return_predictions(self, return_predictions: Optional[bool] = None) -> None:
         # For non ``DDPSpawnPlugin`` plugin, the `return_predictions` is True by default unless user decide otherwise.
         self._return_predictions = not is_ddp_spawn if return_predictions is None else return_predictions
 
+    @property
+    def should_store_predictions(self) -> bool:
+        return self.return_predictions or any(c.interval.on_epoch for c in self.trainer.prediction_writer_callbacks)
+
```

Comment:
Should we merge this with the `return_predictions` getter?

Is there a point to have both?

Reply:
It is slightly cleaner as return_predictions is a user input and could be accessed.

Reply:
And you might want to store the predictions and not return them